lib/valueflow.cpp

```diff
@@ -1011,7 +1011,7 @@ static void setTokenValue(Token* tok, const ValueFlow::Value &value, const Setti
                     (value1.varId == value2.varId && value1.varvalue == value2.varvalue)) {
                     ValueFlow::Value result(0);
                     result.condition = value1.condition ? value1.condition : value2.condition;
-                    result.setInconclusive(value1.isInconclusive() | value2.isInconclusive());
+                    result.setInconclusive(value1.isInconclusive() || value2.isInconclusive());
```

Comment:
Assuming that Value::isInconclusive() is inlined, I *guess* this will just cause extra CPU load. Instead of `A|B` there will be `A ? 1 : B` .. so in best case you replaced a read with a jump (which is probably slower as far as I know).. and in worst case this just means an extra jump.

I think for readability reasons we should allow bitwise or.

Reply:
> Assuming that Value::isInconclusive() is inlined, I _guess_ this will just cause extra CPU load. Instead of `A|B` there will be `A ? 1 : B` .. so in best case you replaced a read with a jump (which is probably slower as far as I know).. and in worst case this just means an extra jump.

True, though the same could be said of many uses of boolean operators where short-circuiting isn't being utilized. I suspect the general practice is to use boolean operators by default, and use bitwise operators in select optimizations.

Reply:
> True, though the same could be said of many uses of boolean operators where short-circuiting isn't being utilized. I suspect the general practice is to use boolean operators by default, and use bitwise operators in select optimizations.

Yes. The checker works as it should if it warns for this code. I just don't want that we use this checker in our own code.

Reply:
I like this PR :-)

I feel I can merge this PR if this change in ValueFlow is reverted.

Reply:
Thanks. :) Reverted the change to ValueFlow.